third_party/nvfuser/csrc/scheduler/utils.cpp

```diff
@@ -2317,6 +2355,124 @@ bool isFastestDimReduction(TensorView* tv) {
   return false;
 }
 
+namespace {
+
+std::vector<TensorView*> getResizedTensors(Fusion* fusion) {
+  std::vector<TensorView*> resized_tensors;
+
+  auto fusion_vals = fusion->usedMathVals();
+  for (auto tv : ir_utils::filterByType<TensorView>(fusion_vals)) {
+    if (ir_utils::hasResizedRfactor(tv)) {
+      resized_tensors.push_back(tv);
+    }
+  }
+
+  return resized_tensors;
+}
+
+} // namespace
+
+void prepareForMemoryTypePromotion(Fusion* fusion) {
+  auto resized_tensors = getResizedTensors(fusion);
+  std::unordered_set<TensorView*> cached;
+  for (auto resized_tensor : resized_tensors) {
+    for (auto producer : ir_utils::producerTvsOf(resized_tensor)) {
+      if (cached.count(producer) != 0) {
+        continue;
+      }
+      producer->cacheAfter();
+      cached.insert(producer);
+    }
+  }
+}
+
+void promoteProducerMemoryTypesOfResizedTensors(Fusion* fusion) {
```

Comment:
Interesting. Can this function be used to do reduction-transpose and persistent-transpose fusion?

Reply:
A similar analysis could work too.